### PR TITLE
move socket.io to shared lib

### PR DESCRIPTION
**Pull request by ivanslabbert**

### CODE REVIEW

**Review by laurensgroeneveld — APPROVED** [review]
Just some small notes mainly on tests, looks fine to me otherwise.

File: shared/src/config/config.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    public readonly appUrl = import.meta.env.VITE_APP_URL;
+    /**
+     * CMS mode flag imported from Vite environment variable VITE_IS_CMS
+     */
+    public readonly isCms = import.meta.env.VITE_IS_CMS;
+    /**
+     * Development mode flag imported from Vite environment variable DEV
+     */
+    public readonly isDevMode = import.meta.env.DEV;
```
[the remark this answers is not in the document]
Comment (ivanslabbert) [reply shared/src/config/config.ts:27]: This needs some rework. It seems that it is not possible for the shared library to read the environmental variables from the CMS / app projects.

File: shared/src/socket/socketio.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    /**
+     * Disconnect and reconnect to the socket server
+     * @param token - Access token
+     */
+    public reconnect(token?: string) {
+        this.socket.disconnect();
+        this._isConnected.value = false;
+        this.socket.auth = { token };
```
[the remark this answers is not in the document]
Comment (ivanslabbert) [reply shared/src/socket/socketio.ts:80]: Agree

File: shared/src/permissions/permissions.ts
[the remark this answers is not in the document]
Comment (ivanslabbert) [reply shared/src/permissions/permissions.ts:1]: Made a separate task for this

File: shared/src/db/database.ts [hunk 3]
```diff
@@ -489,17 +469,100 @@ class database extends Dexie {
...
+        });
     }
 
     /**
-     * Delete a local change by its id
+     * Delete documents to which access has been revoked
      */
-    deleteLocalChange(id: number) {
-        return this.localChanges.where("id").equals(id).delete();
+    private deleteRevoked(accessMap: AccessMap) {
```
[the remark this answers is not in the document]
Comment (ivanslabbert) [reply shared/src/db/database.ts:539]: Added separate task for this

File: shared/src/config/config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivanslabbert) [reply shared/src/config/config.ts]: Global config removed from shared lib

File: cms/src/components/navigation/SideBar.vue [hunk 4]
```diff
@@ -36,26 +43,26 @@ const navigation = ref([
...
+    //     to: { name: "users" },
+    //     icon: UsersIcon,
+    //     visible: hasAnyPermission(DocType.User, AclPermission.View),
+    // },
+    // {
+    //     name: "Groups",
+    //     to: { name: "groups" },
+    //     icon: RectangleStackIcon,
+    //     visible: hasAnyPermission(DocType.Group, AclPermission.View),
+    // },
```
Comment (ivanslabbert) [inline cms/src/components/navigation/SideBar.vue:57]: temporarily disabled until Groups are implemented on new client data model

**Pull request merged**